r/R/dataset-scan.R

```diff
@@ -85,10 +85,22 @@ Scanner$create <- function(dataset,
       # To handle mutate() on Table/RecordBatch, we need to collect(as_data_frame=FALSE) now
       dataset <- dplyr::collect(dataset, as_data_frame = FALSE)
     }
+
+    proj<- c(dataset$selected_columns, dataset$temp_columns)
+    if (!is.null(projection)) {
+      # TODO: should we check names and make sure we only project once per?
+      proj <- c(proj, projection)
+    }
+
+    filt <- dataset$filtered_rows
+    if (!isTRUE(filter)) {
+      filt <- build_expr("&", filt, filter)
+    }
+
```

Comment:
Use this helper to handle a few other edge cases (then restore `dataset$filtered_rows` in the call to Scanner$create below)

```suggestion
    if (!isTRUE(filter)) {
      dataset <- set_filter(dataset, filter)
    }

```